src/corpus/domain/line-details.ts

```diff
@@ -82,12 +86,16 @@ export class LineVariantDetails {
   readonly [immerable] = true
 
   constructor(
-    readonly reconstruction: Token[],
+    readonly reconstruction: readonly Token[],
     readonly note: NoteLine | null,
     readonly manuscripts: readonly ManuscriptLineDisplay[],
     readonly parallelLines: ParallelLine[],
     readonly intertext: MarkupPart[]
-  ) {}
+  ) {
+    if (reconstruction) {
+      this.reconstruction = setSentenceIndices(reconstruction)
+    }
+  }
```

Comment:
will trigger even if list is empty

Reply:
Removing the check causes errors, presumably because src/corpus/application/dtos.ts is not properly typed so `reconstruction` can be `undefined`.